sw.js

```diff
@@ -1,103 +1,47 @@
-  plugins: [
-    new cacheableResponse.CacheableResponsePlugin({ statuses: [0, 200] })
-  ],
-  cacheName
-});
-
-/**
- * Route for local CSS and JS assets
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#stale-while-revalidate
- */
-const assetRoute = new routing.RegExpRoute({
-  regExp: new RegExp(`^${localhost}.*\\.(css|js|png|svg)$`),
-  handler: new runtimeCaching.StaleWhileRevalidate({ requestWrapper })
-});
+// Cache fallback for HTML files
+workbox.routing.registerRoute(
+  /(?:^[^.]+|\.html)$/,
+  workbox.strategies.networkFirst()
+);
 
-/**
- * Route for assets on cloudfront.net CDN
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#stale-while-revalidate
- */
-const cdnAssetRoute = new routing.RegExpRoute({
-  regExp: new RegExp('^https://.*\\.cloudfront\\.net/.*\\.(png|svg)$'),
-  handler: new runtimeCaching.StaleWhileRevalidate({
-    requestWrapper: cdnRequestWrapper
+// Cache CSS and JS files
+workbox.routing.registerRoute(
+  /\.(?:js|css)$/,
+  workbox.strategies.staleWhileRevalidate({
+    cacheName: 'static-resources'
   })
-});
-
-/**
- * Route for pages
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#network-falling-back-to-cache
- */
-const navRoute = new routing.NavigationRoute({
-  whitelist: [/./],
-  handler: new runtimeCaching.NetworkFirst({ requestWrapper })
-});
-
-/**
- * Precache resources
- */
-staticCache.addToCacheList({
-  unrevisionedFiles: ['/', '/404/', '/error/']
-});
-
-self.oninstall = event => {
-  event.waitUntil(Promise.all([staticCache.install().then(skipWaiting)]));
-};
-
-self.onactivate = event => {
-  caches.keys().then(keys => {
-    keys.forEach(cacheName => {
-      if (!cacheName.endsWith(version)) {
-        caches.delete(cacheName);
-      }
-    });
-  });
-  return event.waitUntil(clients.claim());
-};
-
-router.registerRoutes({
-  routes: [assetRoute, cdnAssetRoute, navRoute]
-});
+);
 
-/**
- * Fallback route for stuff that doesn't match
- */
-router.setDefaultHandler({
-  handler: new runtimeCaching.NetworkOnly()
-});
+// Cache images
+workbox.routing.registerRoute(
+  /\.(?:png|gif|jpg|jpeg|svg)$/,
+  workbox.strategies.cacheFirst({
+    cacheName: 'images',
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 60,
+        maxAgeSeconds: 30 * 24 * 60 * 60 // 30 Days
+      })
+    ]
+  })
+);
 
-/**
- * Catch-all fetch error handler
- */
-router.setCatchHandler({
-  handler: {
-    handle: () => caches.match('/error/')
-  }
-});
+// Cache Google Fonts
+workbox.routing.registerRoute(
+  new RegExp('https://fonts.(?:googleapis|gstatic).com/(.*)'),
+  workbox.strategies.cacheFirst({
+    cacheName: 'google-fonts',
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 30
+      }),
+      new workbox.cacheableResponse.Plugin({
+        statuses: [0, 200]
```

Comment:
I'd rather we try not to cache opaque responses (the `0` status). Is there a way to opt-in to CORS mode for the font requests?

Not sure if this helps but just in case: https://developers.google.com/web/tools/workbox/guides/handle-third-party-requests

Reply:
Oh, nice! I just checked and the fonts are already opting into CORS mode via the `crossorigin` attribute. 😉 

```html
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,600,700" crossorigin>
```